charts/flyte-binary/templates/auth-client-secret.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if .Values.configuration.auth.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "flyte-binary.configuration.auth.clientSecretName" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "flyte-binary.labels" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- include "flyte-binary.renderTemplate" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- if .Values.commonAnnotations }}
+    {{- include "flyte-binary.renderTemplate" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+type: Opaque
+stringData:
+  client_secret: {{ required "Internal client secret required when authentication is enabled" .Values.configuration.auth.internal.clientSecret | quote }}
+  oidc_client_secret: {{ required "OIDC client secret required when authentication is enabled" .Values.configuration.auth.oidc.clientSecret | quote }}
```

Comment:
would it make more sense to move this into the one above since this Secret is just for internal use right?  And the other one is all things admin auth?

Reply:
I don't think so. `admin-auth-secret.yaml` is basically only for auto-generated secrets. Anything that we add in there in the helm chart will conflict with the output of the `gen-admin-auth-secret` init container. `auth-client-secret.yaml` is for user-provided secrets, but we're really only storing client secrets in here, so its named accordingly.